innovation.game.php

```diff
@@ -12664,6 +12787,62 @@ function stInterInteractionStep() {
                     }
                     break;
 
+                // id 196, Artifacts age 9: Luna 3
+                case "196N1A":
+                    // Draw and score a card of value equal to the number of cards returned.
+                    self::executeDraw($player_id, $n, 'score');
+                    break;
+
+                // id 199, Artifacts age 9: Philips Compact Cassette
+                case "199N1A":
+                    $colors = self::getValueAsArray(self::getGameStateValue('auxiliary_value'));
+                
+                    foreach ($colors as $color) {
+                        // "Splay up two colors on your board."
+                        self::splay($player_id, $player_id, $color, 3);
+                    }
+                    break;
+
+                // id 200, Artifacts age 9: Syncom 3
+                case "200N1A":
+                    // "Draw and reveal five 9."
+                    $color_detect = array(0,0,0,0,0);
+                    $card_array = array();
+                    $repeat_colors = false;
+                    for($card_idx = 0; $card_idx < 5; $card_idx++) {
+                        $card = self::executeDraw($player_id, 9, 'revealed');
+                        if ($color_detect[$card['color']] == 1) { 
+                            $repeat_colors = true; // if the flag is 1, the color already has been revealed
+                        }
+                        $color_detect[$card['color']] = 1; // flag the color
+                        $card_array[] = $card;
+                    }
+                    
+                    // If you revealed all five colors, you win.
+                    if ($repeat_colors == false) {
+                        self::notifyPlayer($player_id, 'log', clienttranslate('${You} revealed all 5 colors.'), array('You' => 'You'));
+                        self::notifyAllPlayersBut($player_id, 'log', clienttranslate('${player_name} revealed all 5 colors.'), array('player_name' => self::getColoredText(self::getPlayerNameFromId($player_id), $player_id)));
+                        self::setGameStateValue('winner_by_dogma', $player_id);
+                        self::trace('EOG bubbled from self::stPlayerInvolvedTurn Syncom 3');
```

Comment:
This code is in stInterInteractionStep()

self::trace('EOG bubbled from self::stInterInteractionStep Syncom 3');

Reply:
done